inbm/dispatcher-agent/tests/unit/sota/test_os_factory.py

```diff
@@ -33,16 +33,19 @@ def test_raise_error_unsupported_OsFactory(self) -> None:
         self.assertRaises(ValueError, factory.get_os, "MacOS")
 
     def test_create_ubuntu_snapshot_checker(self) -> None:
-        assert type(SotaOsFactory(self.mock_disp_broker, None, []).get_os('Ubuntu').create_snapshotter('update', '1', False)) \
-            is DebianBasedSnapshot
+        ubuntu_os = SotaOsFactory(self.mock_disp_broker, None, []).get_os('Ubuntu')
```

Comment:
**Suggestion:** Refactor the repeated code for creating OS objects into a separate helper method to improve maintainability. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    def get_os_object(self, os_name):
        return SotaOsFactory(self.mock_disp_broker, None, []).get_os(os_name)

    def test_create_ubuntu_snapshot_checker(self) -> None:
        ubuntu_os = self.get_os_object('Ubuntu')
```